controller/controller.ts

```diff
@@ -11,21 +9,20 @@ export class UsersController {
       const users = await TypicodeService.getUsers();
       const transformedUsers = await usersUtil.transformUsers(users);
       return transformedUsers;
-
     } catch (error) {
       console.error('Error calling API:', error);
       throw error;
     }
   }
 
-  async getUserPosts(userId: string): Promise<TransformedPosts[]> {
+  async getUserPosts(userId: string): Promise<transformedNewPost[]> {
     try {
       const posts = await TypicodeService.getPostsByUserId(userId);
-      const transformedPosts = usersUtil.transformUserPosts(posts);
+      const transformedPosts = postsUtil.transformUserPosts(posts);
```

Comment:
You will need to do two calls here: one to get the user information, and the other to get the user post's. So, `transformUserPosts` will need two parameters: one for the user data, and the other for the posts data.